kubetest2/internal/deployers/eksapi/infra.go

```diff
@@ -84,6 +85,31 @@ func (m *InfrastructureManager) createInfrastructureStack(opts *deployerOptions)
 	if err != nil {
 		return nil, err
 	}
+	// get two AZs for the subnets
+	azs, err := m.clients.EC2().DescribeAvailabilityZones(context.TODO(), &ec2.DescribeAvailabilityZonesInput{})
+	if err != nil {
+		return nil, err
+	}
+	var subnetAzs []string
+	if opts.CapacityReservation {
+		subnetAzs, err = m.getAZsWithCapacity(opts)
+		if err != nil {
+			return nil, err
+		}
+		for _, az := range azs.AvailabilityZones {
+			if len(subnetAzs) == 2 {
```

Comment:
could we have a panic if there is only 1 AZ with reserved capacity?

Reply:
We shouldn't, reserved capacity is usually associated with only one AZ.

Reply:
I'm ok with this approach, but it would be better if we just create subnets in All AZs and don't worry about reserved capacity's AZ.